pkg/store/cache/cache.go

```diff
@@ -86,12 +98,34 @@ func (c cacheKey) string() string {
 			key += ":" + c.compression
 		}
 		return key
+	case cacheKeyExpandedPostings:
+		// Use cryptographically hash functions to avoid hash collisions
+		// which would end up in wrong query results.
+		matchers := c.key.(cacheKeyExpandedPostings)
+		matchersHash := blake2b.Sum256([]byte(matchers))
+		key := "EP:" + c.block + ":" + base64.RawURLEncoding.EncodeToString(matchersHash[0:])
```

Comment:
Should we also encode the index header version (https://github.com/thanos-io/thanos/pull/6420/files#diff-3e2896fafa6ff73509c77df2c4389b68828e02575bb4fb78b6c34bcfb922a7ceR2442) because that leads to different results? :thinking:

Reply:
I didn't get it. Block is immutable so since we include the block ID here the version should be the same?

Reply:
Oh I see, you mean we even changed the format of the postings? Like we changed posting list into roaring bitmap or something like that?
Yeah then I think it is a good point.